File: F1Shots/Models/Notification.cs
// Models/Notification.cs

using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

public class Notification
{
    public ObjectId Id { get; set; } // MongoDB ObjectId
    
    [BsonGuidRepresentation(GuidRepresentation.Standard)]
    public Guid Guid { get; set; } // A unique identifier for the notification
    public ObjectId UserId { get; set; } // The recipient of the notification
    public ObjectId SenderUserId { get; set; } // The recipient of the notification
    public ObjectId GroupId { get; set; } // The recipient of the notification
    public string Message { get; set; } // The message content of the notification
    public NotificationType Type { get; set; } // Type of the notification (e.g., FriendRequest, Like, etc.)
    public NotificationStatus Status { get; set; }
    public bool Responded { get; set; }
    public DateTime CreatedAt { get; set; } // Timestamp of when the notification was created
}

public enum NotificationStatus
{
    Unread,   // Notification has not been read yet
    Read,     // Notification has been read
    Archived, // Notification has been archived or dismissed
    Deleted,   // Notification has been deleted (can be used if you want to track deletions)
}


public enum NotificationType
{
    FriendRequest, // For friend requests
    GroupJoinRequest, // For friend requests
    Comment, // For new comments
    Like, // For likes on posts, comments, etc.
    Mention, // For mentions in posts or comments
    GroupInviteRequest, // For group invites
    Info, // For general information notifications
}

File: F1Shots/f1shotsapp/src/Models/Notification.ts
// models/Notification.ts

export interface Notification {
    id: string; // MongoDB ObjectId as string (in frontend, it's usually represented as string)
    userIds: string[]; // The recipient of the notification (ObjectId as string)
    senderUserId: string; // The recipient of the notification (ObjectId as string)
    message: string; // The message content of the notification
    type: NotificationType; // Type of the notification (e.g., FriendRequest, Like, etc.)
    status: NotificationStatus; // Status of the notification (unread, read, etc.)
    createdAt: string; // Timestamp of when the notification was created
}

export enum NotificationStatus {
    Unread = 0,  // Notification has not been read yet
    Read = 1,      // Notification has been read
    Archived = 2, // Notification has been archived or dismissed
    Deleted = 3 // Notification has been deleted (can be used if you want to track deletions)
}

export enum NotificationType {
    FriendRequest = 0, // For friend requests
    GroupJoinRequest = 1, // For friend requests
    Comment = 2,             // For new comments
    Like = 3,                   // For likes on posts, comments, etc.
    Mention = 4             // For mentions in posts or comments
}
